### PR TITLE
Profile: Buy COW token. Waterfall PR[4] 

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Add the `Buy COW` internal Link with the `outputCurrency` parameter using the COW token by chainId.
Screen Shot 2022-04-05 at 18 42 56

## Note
- Might want to set also the `inputCurrency` (USDC?) and an amount for quicker action. Or as initially proposed, checking the user's balance and pick a token (if === USDC/WETH/ETH...etc).

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @fairlighteth , great changes!
However, some my notes: 
1. Buy COW link is always underlined 
image

2. When I press on the Buy COW button, I see the URL is changed. And its parameters aren't cleared out when refresh the page (I can't say that it is an issue)
3. Important: URL parameters aren't cleared out when I change a network, so a user sees 'Select a token' in From and TO fields
url when change network

The same issue is reproducible when copy and paste the link to a different browser

4. As COW Protocol token list was not added to Rinkeby (#363), I see this hanging message when I'm redirected to the SWAP page
рфтпы
It is displayed until I manually press on the Import button. 
(Again, not sure if is an issue)

Thanks!

**Comment by fairlighteth** [comment]
2) When refreshing the page, it will keep the existing URL. I think this is expected behavior. The thing we should be careful about, is if the URL is properly cleared when switching to a different route.

3) This sounds like a valid issue. I think we should reset the URL on network change, in case there are any parameters in the URL. Basically so it routes to the /swap page (for example) and clears out any parameters. Could you create an issue for this and reference this PR?

4) Yes makes sense that this modal is shown. Does the modal work as expected on click of the 'Import' button?

**Comment by elena-zh** [comment]
> Yes makes sense that this modal is shown. Does the modal work as expected on click of the 'Import' button?

**Comment by elena-zh** [comment]
@fairlighteth 
> Yes makes sense that this modal is shown. Does the modal work as expected on click of the 'Import' button?

Yes, it works fine then. I think, the issue will not be reproducible when we will add Cow protocol token list to Rinkeby. 

> This sounds like a valid issue. I think we should reset the URL on network change, in case there are any parameters in the URL. Basically so it routes to the /swap page (for example) and clears out any parameters. Could you create an issue for this and reference this PR?

I have created #386 issue for this. 

Thank you!

**Review by elena-zh — APPROVED** [review]
Approved!

File: src/custom/pages/Profile/index.tsx [hunk 1]
```diff
@@ -228,7 +229,7 @@ export default function Profile() {
 
             <AddToMetamask currency={COW[chainId || 1] as Currency | undefined} />
 
-            <ExtLink href={'#'}>Buy COW ↗</ExtLink>
+            <Link to={`/swap?outputCurrency=${COW_CONTRACT_ADDRESS[chainId || 1]}`}>Buy COW ↗</Link>
```
Comment (W3stside) [inline src/custom/pages/Profile/index.tsx:232]: you'll see these comments a lot in your waterfalls, so i'll just write this one and you change all the subsequent appearances of this. Please use SupportedChainId.MAINNET

**Review by anxolin — COMMENTED** [review]
approve

**Pull request merged**